airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/metadata_assets.py

```diff
@@ -0,0 +1,182 @@
+        "oss": {
+            "enabled": not cloud_only,
+        },
+        "cloud": {
+            "enabled": not oss_only,
+        },
+    }
+
+    if cloud_only or oss_only:
+        return catalogs
+
+    allowed_overrides = [
+        "name",
+        "dockerRepository",
+        "dockerImageTag",
+        "supportsDbt",
+        "supportsNormalization",
+        "license",
+        "supportUrl",
+        "connectionType",
+        "allowedHosts",
+        "normalizationConfig",
+        "suggestedStreams",
+        "resourceRequirements",
+    ]
+
+    # find the difference between the two catalogs
+    for override_col in allowed_overrides:
+        oss_col = override_col + OSS_SUFFIX
+        cloud_col = override_col + CLOUD_SUFFIX
+
+        cloud_value = merged_df.get(cloud_col)
+        oss_value = merged_df.get(oss_col)
+
+        # if the columns are different, add the cloud value to the overrides
+        if cloud_value and not are_values_equal(oss_value, cloud_value):
+            catalogs["cloud"][override_col] = merge_values(oss_value, cloud_value)
+
+    return catalogs
+
+
+def merge_into_metadata_definitions(id_field, connector_type, oss_connector_df, cloud_connector_df) -> pd.Series:
+    merged_connectors = pd.merge(
+        oss_connector_df, cloud_connector_df, on=id_field, how="outer", suffixes=(OSS_SUFFIX, CLOUD_SUFFIX), indicator=True
+    )
+    sanitized_connectors = merged_connectors.where(pd.notnull(merged_connectors), None)
+
+    def build_metadata(merged_df):
+        raw_data = {
+            "name": get_field_with_fallback(merged_df, "name"),
+            "definitionId": merged_df[id_field],
+            "connectorType": connector_type,
+            "dockerRepository": get_field_with_fallback(merged_df, "dockerRepository"),
+            "githubIssueLabel": get_field_with_fallback(merged_df, "dockerRepository").replace("airbyte/", ""),
+            "dockerImageTag": get_field_with_fallback(merged_df, "dockerImageTag"),
+            "icon": get_field_with_fallback(merged_df, "icon"),
+            "supportUrl": get_field_with_fallback(merged_df, "documentationUrl"),
+            "connectionType": get_field_with_fallback(merged_df, "sourceType"),
+            "releaseStage": get_field_with_fallback(merged_df, "releaseStage"),
+            "license": "MIT",
+            "supportsDbt": get_field_with_fallback(merged_df, "supportsDbt"),
+            "supportsNormalization": get_field_with_fallback(merged_df, "supportsNormalization"),
+            "allowedHosts": get_field_with_fallback(merged_df, "allowedHosts"),
+            "normalizationConfig": get_field_with_fallback(merged_df, "normalizationConfig"),
+            "suggestedStreams": get_field_with_fallback(merged_df, "suggestedStreams"),
+            "resourceRequirements": get_field_with_fallback(merged_df, "resourceRequirements"),
+        }
+
+        # remove none values
+        data = {k: v for k, v in raw_data.items() if v is not None}
+
```

Comment:
Doing so allows it to fail the `validate_metadata` with regards to missing keys, right?

Reply:
Basically this is meant to remove optional fields that didn't have a value defined

With out this a lot of connectors metadata files would look like
```yaml
metadata_version: 1.0
data:
    connectorId: abc123
    allowedHosts: null
    supportsNormalization: null
```

Reply:
Which isnt invalid, we would just prefer to if youre not using an optional field to not have it declared

Reply:
That makes sense for things that don't apply to all connectors (e.g. supportsNormalization) - didn't think of that, thought all were required. Makes more sense

Reply:
My only thought is that if there were ones that are technically optional but highly encouraged (allowedhosts, maybe docs url, etc) - having the keys in there with null values would make us more aware that they're missing and should be fixed, versus it being harder to notice when the keys are missing. but with some being completely irrelevant, we probably can't have it both ways - this way works!